### PR TITLE
Add MPU60x0 IMU sensors support for ESP32C3

### PR DESCRIPTION
**Pull request by AuroraRAS**
## Summary
Add MPU60x0 IMU sensors support for ESP32C3

## Impact
ESP32C3

## Testing
ESP32-C3-WROOM-02

### CODE REVIEW

File: boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_bmp180_initialize
```
Comment (acassis) [inline boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c:68]: Please fix function name

File: boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  sninfo("Initializing MPU60X0!\n");
+
+  /* Initialize MPU60X0 */
+
+  i2c = esp32c3_i2cbus_initialize(busno);
+
+  if (i2c)
+    {
+      /* Then try to register the IMU sensor in I2C0 */
```
Comment (acassis) [inline boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c:97]: I2Cx where x = busno

Comment (AuroraRAS) [reply boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c]: I pushed a more compliant version.

File: boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+#include <nuttx/arch.h>
+#include <nuttx/kmalloc.h>
+#include <nuttx/sensors/mpu60x0.h>
+#include <nuttx/i2c/i2c_master.h>
+
+#include "esp32c3_i2c.h"
+
+#include "esp32c3_board_mpu60x0_i2c.h"
+
+/****************************************************************************
```
Comment (pkarashchenko) [inline boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c:39]: I think we can clean-up many unused sections here

File: boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      if (mpu_config == NULL)
+        {
+          snerr("ERROR: Failed to allocate mpu60x0 driver\n");
+          ret = -ENODEV;
+        }
+      else
+        {
+          mpu_config->i2c = i2c;
+          mpu_config->addr = 0x68;
+          snprintf(devpath, 10, "/dev/imu%d", devno);
```
Comment (pkarashchenko) [inline boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c:109]:
```suggestion
          snprintf(devpath, sizeof(devpath), "/dev/imu%d", devno);
```

File: boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c (hunk 3 above)
Comment (AuroraRAS) [reply boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c:38]: It seems necessary in nxstyle.json rules

File: boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c (hunk 4 above)
Comment (AuroraRAS) [reply boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c]: You are right, it looks better now

File: boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c (hunk 3 above)
Comment (pkarashchenko) [reply boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c:38]: The public functions section is obviously needed ;) I was talking about defines, types and other empty sections

Comment (AuroraRAS) [reply boards/risc-v/esp32c3/common/src/esp32c3_board_mpu60x0_i2c.c:38]: Thanks. it's passed the check, but one cancelled after 359 min. timeout maybe?

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**